src/caduceus/outboundSender.go

```diff
@@ -343,70 +343,83 @@ func (obs *CaduceusOutboundSender) Queue(msg *wrp.Message) {
-							if deviceRegex.MatchString(msg.Source) {
+	if false == obs.isValidTimeWindow(now, dropUntil, deliverUntil) {
+		return
+	}
+
+	for _, eventRegex := range events {
+		if false == eventRegex.MatchString(strings.TrimPrefix(msg.Destination, "event:")) {
+			debugLog.Log(logging.MessageKey(),
+				fmt.Sprintf("Regex did not match. got != expected: '%s' != '%s'\n",
+					msg.Destination, eventRegex.String()))
+			continue
+		}
+
+		matchDevice := (nil == matcher)
+		if nil != matcher {
+			for _, deviceRegex := range matcher {
+				if deviceRegex.MatchString(msg.Source) {
+					matchDevice = true
+					break
+				}
+			}
+		}
+		/*
+			// if the device id matches then we want to look through all the metadata
+			// and make sure that the obs metadata matches the metadata provided
+			if matchDevice {
+				for key, val := range metaData {
+					if matchers, ok := matcher[key]; ok {
+						for _, deviceRegex := range matchers {
+							matchDevice = false
+							if deviceRegex.MatchString(val) {
 								matchDevice = true
 								break
 							}
 						}
-					}
-					/*
-						// if the device id matches then we want to look through all the metadata
-						// and make sure that the obs metadata matches the metadata provided
-						if matchDevice {
-							for key, val := range metaData {
-								if matchers, ok := matcher[key]; ok {
-									for _, deviceRegex := range matchers {
-										matchDevice = false
-										if deviceRegex.MatchString(val) {
-											matchDevice = true
-											break
-										}
-									}
-
-									// metadata was provided but did not match our expectations,
-									// so it is time to drop the message
-									if !matchDevice {
-										break
-									}
-								}
-							}
-						}
-					*/
-					if matchDevice {
-						if len(obs.queue) < obs.queueSize {
-							obs.queueDepthGauge.Add(1.0)
-							obs.queue <- msg
-							debugLog.Log(logging.MessageKey(), "WRP Sent to obs queue", "url", obs.id)
-							// a regex was matched, no need to check further matches
+
+						// metadata was provided but did not match our expectations,
+						// so it is time to drop the message
+						if !matchDevice {
 							break
-						} else {
-							obs.queueOverflow()
-							obs.droppedQueueFullCounter.Add(1.0)
 						}
 					}
-				} else {
-					debugLog.Log(logging.MessageKey(),
-						fmt.Sprintf("Regex did not match. got != expected: '%s' != '%s'\n",
-							msg.Destination, eventRegex.String()))
 				}
 			}
-		} else {
-			debugLog.Log(logging.MessageKey(), "Outside delivery window",
-				"now", now, "before", deliverUntil, "after", dropUntil)
-			obs.droppedExpiredCounter.Add(1.0)
+		*/
+		if matchDevice {
+			if len(obs.queue) < obs.queueSize {
+				obs.queueDepthGauge.Add(1.0)
+				obs.queue <- msg
+				debugLog.Log(logging.MessageKey(), "WRP Sent to obs queue", "url", obs.id)
+				// a regex was matched, no need to check further matches
+				break
+			} else {
```

Comment:
this else isn't necessary.

Reply:
I cleaned up the ones in outboundSender.go.  I think it makes sense to do the other files in a different PR.

Reply:
this is outboundSender.go, but more cleanup in a different PR sounds good to me.